### DISCUSSION TITLE
cefSharp.Winform: Could not load file or assembly 'CefSharp.Core.Runtime.DLL' or one of its dependencies. The specified module could not be found

### DISCUSSION THREAD

**Discussion by yashipm** [cefsharp/CefSharp#3758] [General]
Hi,

We have installed CerfSharp using the NuGet Pakage Manager and created a class library which loads a ChromiumWebBrowser, it works fine on VisualStudio-2019,
When we deployed the class library(DLL) into target location along with required cefsharp supported files as mentioned the readme.txt, which throws following exception

Error Message: Could not load file or assembly 'CefSharp.Core.Runtime.DLL' or one of its dependencies. The specified module could not be found.
Stack Trace:    at CefSharp.Cef.get_IsInitialized()

could anyone help us on this issue?

Added following cefsharp supported files in the target deployment location:

* CEF core library.
  * libcef.dll

* Crash reporting library.
  * chrome_elf.dll

* Unicode support data.
  * icudtl.dat

* V8 snapshot data.
  * snapshot_blob.bin
  * v8_context_snapshot.bin

[System Info]:

CefSharp Version=91.1.230.0

OS: Windows 7 x64 

Visual Studio Version:

Microsoft Visual Studio Enterprise 2019
Version 16.7.8
.NET Framework: 4.8

Thanks,

**Comment by ukandrewc** [comment]
If those are the only files you have added, there are quite a few missing.
Check that you've included all the support files from your %project%/bin/debug folder.
My dist folder looks like this: 
![image](https://user-images.githubusercontent.com/30246320/129592042-63b24e5c-cdcf-4a40-8beb-5ed34b2c0090.png)

**Reply by yashipm** [reply]
Thanks for your response,

As mentioned, I have included all the files listed above, still getting the same exception, is there any additional configuration or reference should I add ?

Regards,
Yaship.M

**Comment by ukandrewc** [comment]
@yashipm If you've copied all the files from your debug folder (except .pdb and .xml), then does your target have VC++ 2015 or greater installed?

**Reply by yashipm** [reply]
@ukandrewc : Yes, I copied all the files except .pdp and.xml,  as verified, my target machine has VC++ 2015-2019 Redistributable (X86/x64) 14.27.29114 

Thanks,
Yaship.M

**Comment by ukandrewc** [comment]
I'm new to CEFSharp, but haven't had any issues distributing, can you post an exact list of files in the dist folder.

**Comment by amaitland** [comment]
> When we deployed the class library(DLL) into target location along with required cefsharp supported files as mentioned the readme.txt, which throws following exception

What do you mean by deploy the class library into the target location? What exactly are you trying to achieve? 

> Error Message: Could not load file or assembly 'CefSharp.Core.Runtime.DLL' or one of its dependencies. The specified module could not be found.
Stack Trace: at CefSharp.Cef.get_IsInitialized()

The most common causes are missing visual c++ or missing dependency. 

A detailed list of dependencies is at https://github.com/cefsharp/CefSharp/wiki/Output-files-description-table-%28Redistribution%29
